3_visualize/src/plot_nhdv2_attr.R

```diff
@@ -71,3 +71,55 @@ plot_nhdv2_attr <- function(attr_data,network_geometry,file_path,
 }
 
 
+plot_data_splits <- function(split, split_name, full_df, network_geometry, out_dir, boundary = drb_shp){
+  #' 
+  #' @description This function maps the reaches in the data split as blue
+  #' and the outher reaches in gray.
+  #'
+  #' @param split data frame containing a column for "PRMS_segid"
+  #' @param split_name name of the data split for the plot title
+  #' @param full_df the full dataframe of reaches that have data. Only needs a column for "PRMS_segid"
+  #' @param network_geometry sf object containing the network flowline geometry; 
+  #' must include columns "subsegid" and "geometry"
+  #' @param out_dir a character string that indicates the location of the saved plot
+  #' @param boundary sf polygon defining the boundary of the DRB to plot
+  #'
+  #' @value Returns a png file containing a map
+  
+  #Get the split data to only unique PRMS_segid to be plotted
+  split <- distinct(split, PRMS_segid)
+  full_df <- distinct(full_df, PRMS_segid)
+  
+  attr_plot_spatial <- ggplot() + 
+    #full network (in case there are reaches without data in split or full_df)
+    geom_sf(data = network_geometry, 
+            size = 0.3, color = 'gray') +
+    #DRB boundary
+    geom_sf(data = boundary, fill = NA,
+            size = 0.3, color = 'black') +
+    #full data
+    geom_sf(data = full_df %>% 
+              left_join(.,network_geometry[,c("subsegid","geometry")],
+                        by=c("PRMS_segid"="subsegid")) %>%
+              sf::st_as_sf(),
+            size = 0.3, 
+            color = 'lightblue') +
+    #split data
+    geom_sf(data = split %>% 
+              left_join(.,network_geometry[,c("subsegid","geometry")],
+                        by=c("PRMS_segid"="subsegid")) %>%
+              sf::st_as_sf(),
+            size = 0.3, 
+            color = 'blue') +
```

Comment:
The easiest way to add the colors to a legend is to map the colors onto a plot aesthetic, i.e., by using `aes()` within the layered graphics and then defining the colors using `scale_color_manual`. I think this should give the correct legend, although the plot dimensions probably get wonky: 

```suggestion
    geom_sf(data = network_geometry, 
            aes(color = "no data"),
            size = 0.3) +
    #DRB boundary
    geom_sf(data = boundary, fill = NA,
            size = 0.3, color = 'black') +
    #full data
    geom_sf(data = full_df %>% 
              left_join(.,network_geometry[,c("subsegid","geometry")],
                        by=c("PRMS_segid"="subsegid")) %>%
              sf::st_as_sf(),
            aes(color = "data present in full set"),
            size = 0.3) +
    #split data
    geom_sf(data = split %>% 
              left_join(.,network_geometry[,c("subsegid","geometry")],
                        by=c("PRMS_segid"="subsegid")) %>%
              sf::st_as_sf(),
            aes(color = "data present in split set"),
            size = 0.3) +
    scale_color_manual(name = "",
                       labels = c("no data", "data present in full set", "data present in split set"),
                       values = c("gray", "lightblue", "blue")) + 
```

Reply:
Thanks! This mostly worked, but the layer order on the maps wasn't correct, and it seemed to always plot the "full set" first, regardless of the position in the labels vector. Is there a way to specify the layer order?

Reply:
Oh, right - it plots the layers by alphabetical order. I think in the past I've used the `breaks` argument to make sure the color `values` align with the layers, but if your approach works as you expect I think that's as good an option as any. Good to know!